src/mono/mono/mini/aot-compiler.c

```diff
@@ -5265,6 +5274,20 @@ MONO_RESTORE_WARNING
 		}
 	}
 
+	if (acfg->aot_opts.export_symbols_outfile && export_symbols->len) {
+		char *export_symbols_out = g_string_free (export_symbols, FALSE);
```

Comment:
This should write even if export_symbols->len is 0 so the aot compiler has a fixed set of outputs.

Reply:
Just for my understanding, what is the benefit/need for having a fixed set of outputs? I believe the exportsymbols files are intended to be more internal and not necessarily visible by the user (we plan to delete the export symbols file after successfully building the shared library and linking the export symbols).

Reply:
In general, in a build system you want the build steps to have a consistent set of outputs, so you can specify dependencies between them. Otherwise, the caller has to check whenever the file exists after running the aot compiler, delete it beforehand, etc.

Reply:
Ah, consistent in the sense that if passed the `export-symbols-outfile=` (if in Full AOT Mode), that file will be generated whether or not its empty? The anticipated dependencies with respect to this generated file is to link it in the shared library within CMake, like `"-exported_symbols_list Program.exportsymbols"` on iOS for example.

For the drawback in which the caller needs to check if the file exists, is it also unwanted to add a check on whether or not they're empty when we are linking them?